fv/containers/containers.go

```diff
@@ -109,27 +128,44 @@ func (c *Container) GetHostname() string {
 	return strings.TrimSpace(output)
 }
 
-func (c *Container) WaitRunning(timeout time.Duration) {
+func (c *Container) WaitUntilRunning() {
 	log.Info("Wait for container to be listed in docker ps")
-	start := time.Now()
+
+	// Set up so we detect if container startup fails.
+	stoppedChan := make(chan struct{})
+	go func() {
+		defer close(stoppedChan)
+		err := c.runCmd.Wait()
+		log.WithError(err).WithField("name", c.Name).Info("Container stopped")
+		c.mutex.Lock()
+		defer c.mutex.Unlock()
+		c.runCmd = nil
+	}()
+
 	for {
-		cmd := exec.Command("docker", "ps")
+		Expect(stoppedChan).NotTo(BeClosed())
+
+		cmd := utils.Command("docker", "ps")
 		out, err := cmd.CombinedOutput()
 		Expect(err).NotTo(HaveOccurred())
 		if strings.Contains(string(out), c.Name) {
 			break
 		}
-		if time.Since(start) > timeout {
-			log.Panic("Timed out waiting for container to be listed.")
-		}
+		time.Sleep(1 * time.Second)
```

Comment:
Can we make this sleep smaller?  If it takes about a second to start a container we may add a second to every test.